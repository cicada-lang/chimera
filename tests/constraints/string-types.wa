assert find _ {
  String "A"
}

assert not find _ {
  String 1
}

assert find x {
  x = "cat"
  String x
}

// An unification breaks a type constraint.

assert not find x {
  String x
  x = 1
}

// An unification subsums a type constraint.

assert find x {
  Number x
  x = 1
}
