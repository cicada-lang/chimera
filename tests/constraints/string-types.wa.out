[ _.0 with { String _.0 } ]
